contracts/TokenListManager.sol

```diff
@@ -2,96 +2,217 @@
 pragma solidity ^0.8.20;
 
 import {ITokenListManager} from "./ITokenListManager.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 
 contract TokenListManager is ITokenListManager {
 
     // Storage the token address to whitelist or blacklist
     address[] private whitelistedTokensArray;
     address[] private blacklistedTokensArray;
+    // Declare arrays to store addresses of whitelisted tokens by type
+    address[] private whitelistErc20Tokens;
+    address[] private whitelistErc721Tokens;
+    address[] private whitelistErc1155Tokens;
+    // Declare arrays to store addresses of blacklisted tokens by type
+    address[] private blacklistErc20Tokens;
+    address[] private blacklistErc721Tokens;
+    address[] private blacklistErc1155Tokens;
 
     // Use mapping to check if token in whitelist
     mapping(address => bool) private whitelistedTokens;
     // Use mapping to check if token in blacklist
     mapping(address => bool) private blacklistedTokens;
 
+    // Set permissions
+    modifier onlyOwner() {
+        // Match address
+        /** 0xC0Ec75c25EC201885a791Fd8d39Bf8CE96e1c566
+         * this address is my own test address
+         * owner can use thier own address to replace. */
+        require(msg.sender == 0xC0Ec75c25EC201885a791Fd8d39Bf8CE96e1c566, "Not authorized");
+        _;
+    }
+
     // Implement the addTokens method and add tokens to whitelist
-    function addTokens(address[] calldata tokens) external override {
+    function addTokens(address[] calldata tokens) external override onlyOwner {
```

Comment:
You could refer to the `access` package in openzeppelin lib. There is  `AccessControl` component that is better than `Ownable`.

Reply:
Solved